pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -67,11 +69,19 @@ def done(self) -> bool:
         max_steps_reached = self.max_steps is not None and self.global_step >= self.max_steps
         return max_steps_reached or self.trainer.should_stop or self._num_training_batches_reached(self.is_last_batch)
 
-    def connect(self, trainer: 'pl.Trainer', *args: Any, **kwargs: Any) -> None:
-        """Connects the loop with all necessary parts like trainer and accelerators"""
+    def connect(
+        self,
+        trainer: 'pl.Trainer',
+        *args: Any,
+        progress: Optional[TrainingEpochProgress] = None,
+        **kwargs: Any
+    ) -> None:
+        """Connects the loop with necessary arguments like the trainer"""
         super().connect(trainer, *args, **kwargs)
-        self.batch_loop.connect(trainer)
-        self.val_loop.connect(trainer)
+        if progress is not None:
+            self.progress = progress
+        self.batch_loop.connect(trainer, progress=self.progress.batch, optim_progress=self.progress.optim)
```

Comment:
There's a rough edge here for the optimization progress

On the one hand, we track progress at the epoch level, since we don't care much about the number of optimizer steps etc performed per batch

On the other hand, it is the batch loop who needs to update the counters.

So I would prefer to not pass it down to the next loop here, however, the batch loop would then need to somehow notify the parent loop to update this progress or both should have references to it.

Thoughts?

Reply:
I would avoid adding references to the "parent" loop as long as possible. Such connections could potentially add a lot of confusion when it comes to control flow and nesting, and for debugging. 


> On the other hand, it is the batch loop who needs to update the counters.

which loop exactly is the one that would update the .batch.processed counter? I would say the TrainingEpochLoop, since this is the one who advances from one batch to the next. Don't you think?

Reply:
> which loop exactly is the one that would update the .batch.processed counter? I would say the TrainingEpochLoop, since this is the one who advances from one batch to the next. Don't you think?

Given the initial definition for batch trackers and the current loop structure:

ready: batch fetched from dataloader
started: after on_train_batch_start
processed: after training_step
completed: after on_train_batch_end

Means `ready/completed` would be updated in `TrainingEpochLoop` and `started/processed` in `TrainingBatchLoop`

Reply:
what prevents us from updating the batch progress consistently all in one loop?

Reply:
we can discuss further on the integration PR